### PR TITLE
Update repo structure

### LINKED ISSUE #1: Set up Github Repository
- [x] Create repo
 - [x] Add collaborators 
 - [x] Create project board
 - [x] Create milestone 1
 - [x] Enable Github branch protection
   - Setting > Branches > Branch protection rules and selecting “Add branch protection rule”. Add “main” as the Branch pattern name and tick the box “Require a pull request before merging”. Keep “Require approvals” selected as well. Lock branch. Do not allow bypassing the above settings.
 - [x] Update repository structure
   - `cookiecutter https://github.com/py-pkgs/py-pkgs-cookiecutter.git`
 - [x] Align commit message formats
   - Reference: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#type
 - [x] Set up CD Github action `python-semantic-release`
 - [x] Enable "squash and merge"
 - [x] Create and review github issues

Reference: 
 - https://pages.github.ubc.ca/MDS-2023-24/DSCI_524_collab-sw-dev_book/
 - https://github.ubc.ca/MDS-2023-24/DSCI_532_milestone1_group19/blob/master/milestone1.md

### LINKED ISSUE #11: Repository structure: CODE_OF_CONDUCT.md
to be reviewed: include a citation if you used a template from elsewhere
 - cookicutter
 - DSCI 524
 - python package book

### LINKED ISSUE #12: Repository structure: CONTRIBUTING.md
some information is wrong, to be reviewed

### LINKED ISSUE #14: Repository structure: environment.yml
contain just the basics for now, e.g. the libraries for the dashboard, data wrangling, and vizualization; include version numbers

NOTE: python==3.9.0

### PR DESCRIPTION
**Pull request by JohnShiuMK**
- updated structure to align the requirement: https://github.ubc.ca/MDS-2023-24/DSCI_532_milestone1_group19/blob/master/milestone1.md#4-repository-structure
 - added python-semantic-release github action, that we learn from 524 and I would like to try if it will work here 😀

### CODE REVIEW

File: README.md
Comment (srfrew) [inline README.md:1]: Requirements here are for a project title and summary of purpose. 
I recommend adding TODO for installation until next week.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+### Submit Feedback
+
+If you are proposing a feature:
+
+* Explain in detail how it would work.
+* Keep the scope as narrow as possible, to make it easier to implement.
+* Remember that this is a volunteer-driven project, and that contributions
+  are welcome :)
+
+## Get Started!
```
Comment (srfrew) [inline CONTRIBUTING.md:41]: We do not have installation instructions at this point. Better to replace with "TODO" for the time being.

File: CONTRIBUTING.md
Comment (srfrew) [inline CONTRIBUTING.md:1]: Attribution for the template required at the bottom of this file.

File: LICENSE [hunk 2]
```diff
@@ -1,6 +1,6 @@
 MIT License
```
Comment (srfrew) [inline LICENSE:1]: A non-software license (such as a CC-BY) is needed here to cover non-software content.

File: README.md [hunk 3]
```diff
@@ -1 +1,25 @@
...
+
+## Usage
+
+- TODO
+
+## Contributing
+
+Interested in contributing? Check out the contributing guidelines. Please note that this project is released with a Code of Conduct. By contributing to this project, you agree to abide by its terms.
+
+## License
```
Comment (srfrew) [inline README.md:19]: Update license for non-software portions as recommended in license file comment

File: environment.yml
Comment (srfrew) [inline environment.yml:1]: file needs basic packages to be initialized
(dashboard, data wrangling, visualization, version numbers, jupyter-nb compatibility for development)

File: .github/workflows/ci-cd.yml [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+    # Set up operating system
+    runs-on: ubuntu-latest
+
+    # Define job steps
+    steps:
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.9"
```
Comment (srfrew) [inline .github/workflows/ci-cd.yml]: What python version are we going to use? Here is defaulted as 3.9. 
Could move up to 3.12 if desired, but 3.9 is still supported till 2025

**Review by srfrew — CHANGES_REQUESTED** [review]
Review complete! A few details to be added for repo structure and specific files, see individual comments. Looks great otherwise

**Comment by JohnShiuMK** [comment]
appreciated for the review, @srfrew 
I will update them soon

**Review by srfrew — APPROVED** [review]
All changes match or improve on requested changes, looks awesome!
Approved

**Pull request merged**